utils/pytorch.py

```diff
@@ -172,6 +172,13 @@ def runner_func():
             else:
                 return runner_func()
 
+    def update_last_task_size(self, new_task_size):
+        """
+        A function replacing the last value in the self._workload_size with the new_task_size.
+        Useful for models where the task_size is unknown before finishing the run (eg. text generation models where the number of tokens generated has an upper bound but can be lower)
+        """
+        self._workload_size[-1] = new_task_size
```

Comment:
Works but overall mechanics is not too intuitive now.

I would change this a bit so that:
1. task_size arg in runner.run() function has default value of None
2. if the task size is not specified at runner.run() stage (task size value remains None) and then it's not set by the function here, and the execution proceeds to the next run() call, error is thrown. I.e. task size has to be set either prior or just after each call.
3. runner.run() uses this function as well underneath passing the task size value to it for the convenience of the user. setting task size value both through runner.run(task_size=n) and then through declare_task_size(task_size=n) (my suggested naming in place of update_task_size()) direct call, results in error.
4. all runners (tf, torch, ...) should have this functionality - maybe it can be implemented on root class level?